### DISCUSSION TITLE
How to recombine output triangular mesh to quad mesh

### DISCUSSION THREAD

**Discussion by Gowrimv** [BrunoLevy/geogram#135] [Q&A] [answered]
After remeshing, how do I combine triangular elements to get quads

**Comment by BrunoLevy** [comment]
There is no function to do that in geogram, 

A possible strategy is to compute a score for each triangle edge (depending on both co-planarity of the triangles and angles at the four corners). Then you can try a greedy strategy (sort all edges by score, then merge the best ones in order).

There is also a more complicated algorithm (called "perfect matching")  that will maximize the total score. It is used in the following article:
https://ui.adsabs.harvard.edu/abs/2012IJNME..89.1102R/abstract

**Comment by Gowrimv** [comment]
Hello,
We had used blossom quad algorithm, but without CVT. So it was not giving
desired results. So geogram can only generate triangular mesh?
Gowri

**Comment by BrunoLevy** [comment, marked as answer]
There exists a quad and hex mesher in the "Vorpaline/geogram+" extension packages marketed by the 
[Tessael](https://www.tessael.com/) company.

**Comment by Gowrimv** [comment]
Okay. Thanks for the tip!
